asteroids_code/BoilerPlate/player.cpp

```diff
@@ -10,27 +10,33 @@ void Player::Render()
 	//put a thrusting figure (like a fire) at the back of the spaceship
 	glLoadIdentity();
 	glTranslatef(origin.x, origin.y, 0.0);
+	glRotatef(angle_degree, 0.0, 0.0, 1.0);
 	Ship_Render();
+	if (thruster_on)
+	{
+		Thruster_Render();
+		thruster_on = false;
+	}
+
 }
 
 void Player::Ship_Render()
 {
 	glBegin(GL_LINE_LOOP);
-	glVertex2f(0.0, 30.0);
-	glVertex2f(-10.0, 0.0);
-	glVertex2f(-25.0, -10.0);
-	glVertex2f(-35.0, -25.0);
-	glVertex2f(-25.0, -25.0);
-	glVertex2f(-10.0, -20.0);
-	glVertex2f(-5.0, -10.0);
-	glVertex2f(-2.5, -25.0);
-	glVertex2f(2.5, -25.0);
-	glVertex2f(5.0, -10.0);
-	glVertex2f(10.0, -20.0);
-	glVertex2f(25.0, -25.0);
-	glVertex2f(35.0, -25.0);
-	glVertex2f(25.0, -10.0);
-	glVertex2f(10.0, 0.0);
+	for (int x = 0; x < ship_points.size();x++)
```

Comment:
* Spacing should be consistent, the update part of the for does not match the initialize and the loop condition.
* Take a look at iterators, which are more convenient when using STL containers.